### PR TITLE
Adding release notes, and highlights for 2.3.0 release

### PR DESCRIPTION
**Pull request by barkbay**
Release notes and highlights for the 2.3.0 release.

### CODE REVIEW

File: docs/release-notes/highlights-2.3.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  username: myuser
+  password: mypassword # clear text value of the password
+  roles: kibana_admin,ingest_admin  # optional
+----
+
+[float]
+[id="{p}-230-observedGeneration-"]
+==== Observed generation
+
+The last resource's generation processed by the operator, also known as the "observed generation", is now set and maintained in the status subresource on all kinds of Elastic resources. While in version 2.2 this enhancement was only available on Elasticsearch and Kibana, it is now also enabled on Agent, Beats, APM Server and Enterprise Search resources.
```
Comment (pebrc) [inline docs/release-notes/highlights-2.3.0.asciidoc:33]:
```suggestion
The most recent generation of a resource processed by the operator, also known as the "observed generation", is now set and maintained in the status subresource on all kinds of Elastic resources. While in version 2.2 this enhancement was only available on Elasticsearch and Kibana, it is now also enabled on Agent, Beats, APM Server and Enterprise Search resources.
```

File: docs/release-notes/highlights-2.3.0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[[release-highlights-2.3.0]]
+== 2.3.0 release highlights
+
+[float]
+[id="{p}-230-new-and-notable"]
+=== New and notable
+
+New and notable changes in version 2.3.0 of {n}. Check <<release-notes-2.3.0>> for the full list of changes.
+
```
Comment (pebrc) [inline docs/release-notes/highlights-2.3.0.asciidoc:9]: Should we mention that we are going back to distroless as the base image ?

**Review by pebrc — APPROVED** [review]

File: docs/release-notes/2.3.0.asciidoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+[[docs-2.3.0]]
+[float]
+=== Documentation improvements
+
+* [a11y] Fix "below" occurrences {pull}5714[#5714] (issue: {issue}5306[#5306])
+* [a11y] Fix "above" occurrences {pull}5672[#5672] (issue: {issue}5306[#5306])
+* Change references to the master branch to main in the CONTRIBUTING guide. {pull}5741[#5741]
+* Fix Helm command examples in docs {pull}5737[#5737]
+* Update list of ECK versions that triggers a rolling restart {pull}5715[#5715] (issue: {issue}5648[#5648])
+* Update doc pages that use the repository-gcs plugin {pull}5700[#5700] (issue: {issue}5457[#5457])
```
Comment (alaudazzi) [inline docs/release-notes/2.3.0.asciidoc:49]:
```suggestion
* Update documentation pages that use the repository-gcs plugin {pull}5700[#5700] (issue: {issue}5457[#5457])
```

File: docs/release-notes/2.3.0.asciidoc [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+* CronJob batch/v1beta1 no longer served in 1.25 {pull}5685[#5685]
+* Update doc to customize pods {pull}5660[#5660]
```
Comment (alaudazzi) [inline docs/release-notes/2.3.0.asciidoc:51]:
```suggestion
* Update documentation to customize pods {pull}5660[#5660]
```

File: docs/release-notes/2.3.0.asciidoc [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+** Elastic Maps {pull}5536[#5536] (issue: {issue}3392[#3392])
+** EnterpriseSearch {pull}5526[#5526] (issue: {issue}3392[#3392])
+** Elastic APM Server {pull}5470[#5470] (issue: {issue}3392[#3392])
+* Upgrade PodDisruptionBudget from v1beta1 to v1 {pull}5709[#5709]
+* Support disable-downgrade-validation for all relevant apps {pull}5680[#5680] (issue: {issue}5531[#5531])
+* Allow non-IPs in service spec to avoid noop updates {pull}5663[#5663] (issue: {issue}5657[#5657])
+* Add APM tracing for client-go requests to the Kubernetes API {pull}5651[#5651]
+* Add support for the desired nodes API {pull}5650[#5650]
+* Base ECK docker image on distroless instead of UBI by default  {pull}5580[#5580] (issue: {issue}4561[#4561])
+* Added priority class and leader election to operator helm chart {pull}5538[#5538]
```
Comment (alaudazzi) [inline docs/release-notes/2.3.0.asciidoc:30]:
```suggestion
* Added priority class and leader election to operator Helm chart {pull}5538[#5538]
```

**Review by alaudazzi — APPROVED** [review]
A few minor edits, otherwise LGTM.

**Pull request merged**